### PR TITLE
Commit in staged/unstaged menu

### PR DESCRIPTION
**Pull request by arnaudperalta**
- **PR Description**
3 new commit actions in staged/unstaged menu.
Closes https://github.com/jesseduffield/lazygit/issues/2285

- **Please check if the PR fulfills these requirements**

* [x] Cheatsheets are up-to-date (run `go run scripts/cheatsheet/main.go generate`)
* [x] Code has been formatted (see [here](https://github.com/jesseduffield/lazygit/blob/master/CONTRIBUTING.md#code-formatting))
* [x] Tests have been added/updated (see [here](https://github.com/jesseduffield/lazygit/blob/master/pkg/integration/README.md) for the integration test guide)
* [x] Text is internationalised (see [here](https://github.com/jesseduffield/lazygit/blob/master/CONTRIBUTING.md#internationalisation))
* [x] Docs (specifically `docs/Config.md`) have been updated if necessary
* [x] You've read through your own file changes for silly mistakes etc

### CODE REVIEW

**Comment by arnaudperalta** [comment]
Does this PR needs new integration tests ?

**Comment by mark2185** [comment]
> Does this PR needs new integration tests ?

Well, it would be nice :)

Just so we know for sure that everything works as expected, and this seems testable. At least for the non-"open-up-external-editor" cases.

**Comment by arnaudperalta** [comment]
> > Does this PR needs new integration tests ?
> 
> Well, it would be nice :)
> 
> Just so we know for sure that everything works as expected, and this seems testable. At least for the non-"open-up-external-editor" cases.

I just wrote two integration tests, one who commits in the staged files menu and the other who commits in the unstaged one. Should I add two more with commit without pre-hook ?

File: pkg/integration/tests/commit/staged.go [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+)
+
+var Staged = NewIntegrationTest(NewIntegrationTestArgs{
+	Description:  "Staging a couple files, going in the staged files menu and committing",
+	ExtraCmdArgs: "",
+	Skip:         false,
+	SetupConfig:  func(config *config.AppConfig) {},
+	SetupRepo: func(shell *Shell) {
+		shell.CreateFile("myfile", "myfile content")
+		shell.CreateFile("myfile2", "myfile2 content")
```
Comment (mark2185) [inline pkg/integration/tests/commit/staged.go:15]: You can chain these, e.g.
```
	shell.
             CreateFile("myfile", "myfile content").
	     CreateFile("myfile2", "myfile2 content")
```

Just to be consistent with other tests.

File: pkg/integration/tests/commit/staged.go [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+		input.PrimaryAction()
+		input.Confirm()
+		input.PressKeys(keys.Files.CommitChanges)
+
+		commitMessage := "my commit message"
+		input.Type(commitMessage)
+		input.Confirm()
+
+		assert.CommitCount(1)
+		assert.MatchHeadCommitMessage(Equals(commitMessage))
```
Comment (mark2185) [inline pkg/integration/tests/commit/staged.go:31]: We should also assert that we're still in the `Staged panel` after it's done.

File: pkg/integration/tests/commit/unstaged.go [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+		input.NextItem()
+		input.Confirm()
+		input.PressKeys(keys.Files.CommitChanges)
+
+		commitMessage := "my commit message"
+		input.Type(commitMessage)
+		input.Confirm()
+
+		assert.CommitCount(1)
+		assert.MatchHeadCommitMessage(Equals(commitMessage))
```
Comment (mark2185) [inline pkg/integration/tests/commit/unstaged.go:30]: We should also assert that we're still in the `Unstaged panel` after it's done.

**Review by mark2185 — CHANGES_REQUESTED** [review]
Just a few things :)

**Comment by mark2185** [comment]
> Should I add two more with commit without pre-hook ?

I'd rather actually that we rework the whole "run commit without hooks" with the whole `WIP` prefix because that seems rather hacky, so I'm not sure if we should write tests for it just yet. Summoning @jesseduffield .

Also, I'd like a test that stages a single line, commits it, then stages another, and commits that because that's a real use case. The current tests do look like they're doing what one would expect, but it's even nicer if the tests look like something a real user would do :)

**Comment by jesseduffield** [comment]
The run commit without hooks thing is a little hacky but I think it's still worth having a test for it because the hackiness that we would want to fix shouldn't break the test. That is: currently when we press 'W' the commit message panel comes up and if you hit enter on that, so long as the 'WIP' prefix is still there in the text area it'll commit without invoking hooks. What I want to change is to make it so that when you press 'w' it will skip hooks no matter what content you write (but I'd still have the 'WIP' prefix there to start with). So I think a test is warranted.

**Comment by arnaudperalta** [comment]
I wrote two new tests with the pre-commit hook commit feature.
Tell me if I need to add a proper hook for these tests.

**Comment by arnaudperalta** [comment]
Any feedbacks @mark2185 @jesseduffield ? And is this normal that the CI is sometimes failing ?

**Comment by jesseduffield** [comment]
code all looks good @arnaudperalta . I suspect the failing integration test is a legit failure. I've just done a rebase onto master so we'll see if it fails again

**Comment by arnaudperalta** [comment]
> code all looks good @arnaudperalta . I suspect the failing integration test is a legit failure. I've just done a rebase onto master so we'll see if it fails again

Thanks, seems good now.

**Comment by jesseduffield** [comment]
Nice one, thanks for making this @arnaudperalta !

**Comment by ooloth** [comment]
Excited for this one to go out. 🙌

**Pull request merged**